toffy/rosetta.py

```diff
@@ -68,18 +72,79 @@ def compensate_matrix_simple(raw_inputs, comp_coeffs):
     # set negative values to zero
     outputs = np.where(outputs > 0, outputs, 0)
 
+    # subset on specified indices
+    outputs = outputs[..., out_indices]
+
     return outputs
 
 
-def compensate_image_data(raw_data_dir, comp_data_dir, comp_mat_path, panel_info_path,
-                          save_format='normalized', batch_size=10, gaus_rad=1, norm_const=100):
+def validate_inputs(raw_data_dir, comp_mat, acquired_masses, acquired_targets, input_masses,
+                    output_masses, all_masses, fovs, save_format, batch_size, gaus_rad):
+    """Helper function to validate inputs for compensate_image_data
+
+    Args:
+        raw_data_dir (str): path to raw data
+        comp_mat (pd.DataFrame): compensation matrix
+        acquired_masses (list): masses in the supplied panel
+        acquired_targets (list): targets in the supplied panel
+        input_masses (list): masses to use for compensation
+        output_masses (list): masses to compensate
+        all_masses (list): masses in the compensation matrix
+        fovs (list): fovs in the raw_data_dir
+        save_format (str): format to save the data
+        batch_size (int): number of images to process concurrently
+        gaus_rad (int): radius for smoothing"""
+
+    # make sure panel is in increasing order
+    if not np.all(acquired_masses == sorted(acquired_masses)):
+        raise ValueError("Masses must be sorted numerically in the panel_info file")
```

Comment:
This works, an alternative method is to use `verify_same_elements`, which now has an `enforce_order` argument that you can set to `True` (added as part of the pixel/cell clustering pipeline).